lucien-ios/AddNewComicBook/AddComicViewController.swift

```diff
@@ -276,39 +314,43 @@ final class AddComicViewController: UIViewController, UIPickerViewDelegate, UIPi
     }
 
     func textFieldShouldReturn(_ textField: UITextField) -> Bool {
-        if let nextTextField = textField.superview?.superview?.viewWithTag(textField.tag + 1) as? UITextField {
-            nextTextField.becomeFirstResponder()
-        } else {
-            textField.resignFirstResponder()
-        }
+        guard let textFields = addComicViewControllerTextFields else { return false }
+        let nextTextField = textFields[textField.tag + 1]
```

Comment:
Doesn't this crash if you hit enter on the last text field?

Either check the length before this, or use `ip_safely` subscripting from the Intrepid pod.

Also, to remove a reliance on tags from your .xib being correct, you could find the index of `textField` in the array and then grab the next one if it exists.

Reply:
It does not crash. The reason why it doesn't crash is because the last textfield has a keypad keyboard which does not have a `return` key. It has a custom "Done" button toolbar above it that calls `resignFirstResponder`.

Reply:
For best practice and possible crashes (edits in the future etc..) I'll change the logic to check for length.

Reply:
Seconding Paul's comment here, we shouldn't be relying on UI (which can change) to prevent crashes, and we shouldn't be using the tags to identify text fields.

Reply:
I tried to work with the Intrepid pod, but I decided to check for length instead due to errors like these: 

![screen shot 2017-10-10 at 4 53 53 pm](https://user-images.githubusercontent.com/3248926/31410956-20363f9a-adde-11e7-8f5a-3b90ff6447b1.png)